base-notebook/start.sh

```diff
@@ -111,6 +111,13 @@ else
     if [[ "$GRANT_SUDO" == "1" || "$GRANT_SUDO" == 'yes' ]]; then
         echo 'Container must be run as root to grant sudo permissions'
     fi
+    for f in /usr/local/bin/start-notebook.d/*; do
+      case "$f" in
+        *.sh)     echo "$0: running $f"; . "$f" ;;
+        *)        echo "$0: ignoring $f" ;;
+      esac
+      echo
+    done
```

Comment:
Do we want this logic to execute if the user runs the container with `--user root` too? As written, it only runs in the `else` block of the surrounding check for the container user ID.

Reply:
good point. I think this should be moved to the top out of the `if` doing uid check.  With that we can gradually refactor this giant if block into smaller scripts.